### PR TITLE
Adds additional handler to support subscription success notifications

### PR DESCRIPTION
**Pull request by nihalbhatnagar**

### CODE REVIEW

File: packages/api/src/experimental/subscribe.ts [hunk 1]
```diff
@@ -22,21 +22,23 @@ import type {
 import type { EXPERIMENTAL_ObjectSetListener } from "../public/unstable.js";
 import type { Experiment } from "./Experiment.js";
 
+type NonEmptyArray<T> = [T, ...T[]];
```
Comment (ericanderson) [inline packages/api/src/experimental/subscribe.ts:25]: I'm 80% sure we shouldn't use this type. I've seen this type show up in other code bases where it just ends up proliferating all over the place causing more harm than good.

File: packages/client/src/objectSet/ObjectSetListenerWebsocket.ts [hunk 2]
```diff
@@ -451,7 +422,7 @@ export class ObjectSetListenerWebsocket {
     payload: ObjectSetUpdates,
   ) => {
     const sub = this.#subscriptions.get(payload.id);
-    invariant(sub, `Expected subscription id ${payload.id}`);
+    if (sub == null) return;
```
Comment (ericanderson) [inline packages/client/src/objectSet/ObjectSetListenerWebsocket.ts:425]: The old code assumed you would always be able to find by payload id. Is this no longer the case? Why?

File: packages/client/src/objectSet/ObjectSetListenerWebsocket.ts [hunk 3]
```diff
@@ -323,16 +299,11 @@ export class ObjectSetListenerWebsocket {
...
     sub.status = newStatus;
     // make sure listeners do nothing now
     sub.listener = fillOutListener<Q, any>({});
-    if (sub.expiry) {
-      clearTimeout(sub.expiry);
-      sub.expiry = undefined;
-    }
     this.#subscriptions.delete(sub.subscriptionId);
-
+    this.#sendSubscribeMessage();
```
Comment (ericanderson) [inline packages/client/src/objectSet/ObjectSetListenerWebsocket.ts:306]: Why was this added here?

Comment (nihalbhatnagar) [reply packages/client/src/objectSet/ObjectSetListenerWebsocket.ts:306]: To unsubscribe from an object set, you need to send another subscribe request without the given subscription request

File: packages/client/src/objectSet/ObjectSetListenerWebsocket.ts (hunk 2 above)
Comment (nihalbhatnagar) [reply packages/client/src/objectSet/ObjectSetListenerWebsocket.ts:425]: I forgot to test this, but at least in the new code when we delete the subscription in unsubscribe, without necessarily unsubscribing by sending a new request, this would error because we still received updates for that sub id. Even though we are unsubscribing and this should not happen, I wanted to remove this as a failsafe in case we still receive an update here

File: packages/api/src/experimental/subscribe.ts (hunk 1 above)
Comment (nihalbhatnagar) [reply packages/api/src/experimental/subscribe.ts:25]: I see, how should we enforce a non empty array here then? Does it make sense to throw if we receive an empty array? I don't want to accept empty arrays because that will then request all non reference backed properties and some users might become reliant on that. Also curious to know what issues it creates

File: packages/client/src/objectSet/ObjectSetListenerWebsocket.ts (hunk 2 above)
Comment (ericanderson) [reply packages/client/src/objectSet/ObjectSetListenerWebsocket.ts:425]: Okay

File: packages/api/src/experimental/subscribe.ts (hunk 1 above)
Comment (nihalbhatnagar) [reply packages/api/src/experimental/subscribe.ts]: Discussed offline, will add a throw here on a FLUP

File: packages/client/src/objectSet/ObjectSetListenerWebsocket.ts (hunk 3 above)
Comment (ericanderson) [reply packages/client/src/objectSet/ObjectSetListenerWebsocket.ts:306]: I see, so the old code kept getting updates (that were dropped because we reset the listener to noop) but this pre-emptively unsubscribes to stop the network traffic.

File: packages/api/src/experimental/subscribe.ts [hunk 4]
```diff
@@ -22,14 +22,12 @@ import type {
...
 
-type NonEmptyArray<T> = [T, ...T[]];
-
 type subscribeFn = <
   Q extends ObjectOrInterfaceDefinition,
   const P extends PropertyKeys<Q>,
 >(
   objectSet: ObjectSet<Q>,
-  properties: NonEmptyArray<P>,
+  properties: Array<P>,
```
Comment (ericanderson) [inline packages/api/src/experimental/subscribe.ts:30]: Probably should make this a `ReadOnlyArray<P>` too now that I think about it, so that people who have those in their redux can use it

**Review by ericanderson — APPROVED** [review]
Feel free to do followup if you want

File: packages/api/src/experimental/subscribe.ts (hunk 4 above)
Comment (nihalbhatnagar) [reply packages/api/src/experimental/subscribe.ts:30]: Will flup this with the other changes
